assets/js/theme/global/compare-products.js

```diff
@@ -32,6 +34,12 @@ function decrementCounter(itemID) {
         const url = new URL(pathname)
         window.history.replaceState(null, '', url)
         q$(`[${selectors.dataCompareCard}='${itemID}']`).remove()
+
+        if (compareProductsIDs.length === 0) {
```

Comment:
This code isn't strictly related to the `decrementCounter` function. Let's move it into a dedicated one called `handleNoResults` (or something like that). Also, handling visibility by CSS seems easier - as you wouldn't need to select multiple elements via js, and update their styles individually. 

So, how about adding a class "no-results" to the container, and based on this, display or hide compare products / remove all button and a proper content for h1? 🤔